src/universal/components/SignInPage/SignInPage.js

```diff
@@ -1,107 +1,77 @@
 /**
- * The sign-in page.  Hosts 3rd part signin, email/password signin, and
- * also functions as the callback handler for the Auth0 OIDC response.
+ * The sign-in page.  Hosts 3rd party signin, email/password signin, and also
+ * functions as the callback handler for the Auth0 OIDC response.
  *
  * @flow
  */
-import type {Dispatch} from 'redux';
+
+import type {Node} from 'react';
 import type {RouterHistory, Location} from 'react-router-dom';
+import type {Dispatch} from 'redux';
+import type {AuthResponse, Credentials} from 'universal/types/auth';
 
-import {WebAuth} from 'auth0-js';
-import React, {Component} from 'react';
-import styled from 'react-emotion';
+import React, {Component, Fragment} from 'react';
 import {connect} from 'react-redux';
 import {Link, withRouter} from 'react-router-dom';
 
 import signinAndUpdateToken from 'universal/components/Auth0ShowLock/signinAndUpdateToken';
+import AuthPage from 'universal/components/AuthPage/AuthPage';
 import LoadingView from 'universal/components/LoadingView/LoadingView';
+import loginWithToken from 'universal/decorators/loginWithToken/loginWithToken';
 import withAtmosphere from 'universal/decorators/withAtmosphere/withAtmosphere';
+import auth0Login from 'universal/utils/auth0Login';
+import {THIRD_PARTY_AUTH_PROVIDERS} from 'universal/utils/constants';
+import getWebAuth from 'universal/utils/getWebAuth';
 
-import Header from './Header';
 import SignIn from './SignIn';
 
-type Credentials = {
-  email: string,
-  password: string
-};
-
-type ParsedAuthResponse = {
-  idToken: string,
-  idTokenPayload: Object
-};
-
 type Props = {
   atmosphere: Object,
   dispatch: Dispatch<*>,
-  hasSession: boolean,
   history: RouterHistory,
   location: Location,
   webAuth: Object
 };
 
 type State = {
   loggingIn: boolean,
-  error: ?Error
+  error: ?{error_description: string},
```

Comment:
if we only use the `error_description` can we just set that as the error so we only handle a string?
if not can we make it our own & camelCase it? (this should be an eslint rule, not sure why it's not picking it up...)

Reply:
Yeah, good catch!